src/Renderer/ThreeExtended/PlanarControls.js

```diff
@@ -0,0 +1,762 @@
+/** Description: Camera controls adapted for a planar view, with animated movements
+* Left mouse button : "drag" the ground, translating the camera on the (xy) world plane.
+* Right mouse button : translate the camera on local x and world z axis (pan)
+* Ctrl + left mouse : rotate (orbit) around the camera's focus point.
+* Scroll wheel : zooms toward cursor position (animated).
+* Middle mouse button (wheel click) : 'smart zoom' at cursor location (animated).
+* S : go to start view (animated)
+* T : go to top view (animated)
+* How to use : instanciate PlanarControls after camera setup (setPosition and lookAt)
+* or you can also setup the camera with options.startPosition and options.startLookAt
+*/
+
+import * as THREE from 'three';
+
+// event keycode
+const keys = { CTRL: 17, SPACE: 32, S: 83, T: 84 };
+const mouseButtons = { LEFTCLICK: THREE.MOUSE.LEFT, MIDDLECLICK: THREE.MOUSE.MIDDLE, RIGHTCLICK: THREE.MOUSE.RIGHT };
+
+// control state
+const STATE = { NONE: -1, DRAG: 0, PAN: 1, ROTATE: 2, TRAVEL: 3 };
+let state = STATE.NONE;
+let isCtrlDown = false;
+
+// eventListeners handlers
+let _handlerOnKeyDown;
+let _handlerOnKeyUp;
+let _handlerOnMouseDown;
+let _handlerOnMouseUp;
+let _handlerOnMouseMove;
+let _handlerOnMouseWheel;
+
+// mouse movement
+const mousePosition = new THREE.Vector2();
+const lastMousePosition = new THREE.Vector2();
+const deltaMousePosition = new THREE.Vector2(0, 0);
+
+// drag movement
+const dragStart = new THREE.Vector3();
+const dragEnd = new THREE.Vector3();
+const dragDelta = new THREE.Vector3();
+
+// camera focus point : ground point at screen center
+const centerPoint = new THREE.Vector3(0, 0, 0);
+
+// camera rotation
+let phi = 0.0;
+let thetaDelta = 0;
+let phiDelta = 0;
+
+// animated travel
+let travelStarted = false;
+const travelEndPos = new THREE.Vector3();
+const travelStartPos = new THREE.Vector3();
+const travelStartRot = new THREE.Quaternion();
+const travelEndRot = new THREE.Quaternion();
+let travelAlpha = 0;
+let travelDuration = 0;
+let travelUseRotation = false;
+let travelUseSmooth = false;
+
+// time management
+let deltaTime = 0;
+let lastElapsedTime = 0;
+const clock = new THREE.Clock();
```

Comment:
How about storing all these inside the `PlanarControls` instance so a user can instanciate several controls?